assets/queries/openAPI/general/invalid_schema_external_documentation_url/query.rego

```diff
@@ -0,0 +1,57 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	version := openapi_lib.check_openapi(doc)
+	version != "undefined"
+
+	[path, value] = walk(doc)
+	url := value.schema.externalDocs.url
+	not openapi_lib.is_valid_url(url)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.schema.externalDocs.url", [openapi_lib.concat_path(path)]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
+		"keyActualValue": "Schema External Documentation URL is not a valid URL",
+		"overrideKey": version,
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	version := openapi_lib.check_openapi(doc)
+	version == "3.0"
+
+	url := doc.components.schemas[s].externalDocs.url
+	not openapi_lib.is_valid_url(url)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("components.schemas.{{%s}}.externalDocs.url", [s]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
+		"keyActualValue": "Schema External Documentation URL is not a valid URL",
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	version := openapi_lib.check_openapi(doc)
+	version == "2.0"
+
+	url := doc.definitions[s].externalDocs.url
+	not openapi_lib.is_valid_url(url)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("definitions.{{%s}}.externalDocs.url", [s]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
+		"keyActualValue": "Schema External Documentation URL is not a valid URL",
+		"overrideKey": version,
+	}
+}
```

Comment:
I think this to rules can be merged as following:
```suggestion
CxPolicy[result] {
	doc := input.document[i]
	version := openapi_lib.check_openapi(doc)
	def := get_schema_definitions(doc, version)

	url := def[s].externalDocs.url
	not openapi_lib.is_valid_url(url)

	result := {
		"documentId": doc.id,
		"searchKey": sprintf("%s.{{%s}}.externalDocs.url", [get_schema_definitions_path(version), s]),
		"issueType": "IncorrectValue",
		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
		"keyActualValue": "Schema External Documentation URL is not a valid URL",
		"overrideKey": version,
	}
}

get_schema_definitions(docs, version) = obj {
  version == "3.0"
  obj := doc.components.schemas
} else = obj {
  version == "2.0"
  obj := doc.definitions
}

get_schema_definitions_path(version) = ret {
  version == "3.0"
  ret := "components.schemas"
} else = obj {
  version == "2.0"
  ret := "definitions"
}
```

Reply:
A nice suggestion 🚀 Thank you! We can merge `get_schema_definitions` and `get_schema_definitions_path`:

```
get_schema_info(doc, version) = schemaInfo {
	version == "3.0"
	schemaInfo := {"obj": doc.components.schemas, "path": "components.schemas"}
} else = schemaInfo {
	version == "2.0"
	schemaInfo := {"obj": doc.definitions, "path": "definitions"}
}
```